ch-05/src/SOlution516.java

```diff
@@ -0,0 +1,39 @@
+import java.util.Scanner;
+
+public class SOlution516 {
+    public static void main(String[] args) {
+        final Scanner input = new Scanner(System.in);
+        int number = input.nextInt();
+        int n = 1;
+
+        if (number > 1) {
+            while (number != 1) {
+                if (isPrime(n)) {
+                    if (number % n == 0) {
+                        System.out.printf("%d ", n);
+                        number /= n;
+                        n = 1;
+                    }
+                }
+                n++;

```

Comment:
Here, you take a few more steps than needed: every time, you find a factor, you reset `n` to `1`. However, since you're starting from `1`, if you find a factor `n`, then for any other factor `x` you will have `x >= n`. 

Thus, you can reduce the number of iterations, by keeping the value of `n`:

~~~~Java
if (isPrime(n) && (number % n == 0)) {
  System.out.printf("%d ", n);
  number /= n;
} else {
  n++;
}
~~~~